backend/src/test/java/codezap/tag/repository/TagJpaRepositoryTest.java

```diff
@@ -0,0 +1,129 @@
+package codezap.tag.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import codezap.global.exception.CodeZapException;
+import codezap.global.repository.JpaRepositoryTest;
+import codezap.tag.domain.Tag;
+
+@JpaRepositoryTest
+class TagJpaRepositoryTest {
+
+    @Autowired
+    private TagRepository tagRepository;
+
+    @Nested
+    @DisplayName("id로 태그 조회")
+    class fetchById {
+        @Test
+        @DisplayName("성공 : id로 태그를 알아낼 수 있다.")
+        void fetchByIdSuccess() {
+            Tag tag = tagRepository.save(new Tag("tag"));
+
+            Tag actual = tagRepository.fetchById(tag.getId());
+
+            assertThat(actual).isEqualTo(tag);
+        }
+
+        @Test
+        @DisplayName("실패 : 존재하지 않는 id인 경우 에러가 발생한다.")
+        void fetchByIdFailByNotExistsId() {
+            long notExistId = 100;
+
+            assertThatThrownBy(() -> tagRepository.fetchById(notExistId))
+                    .isInstanceOf(CodeZapException.class)
+                    .hasMessage("식별자 " + notExistId + "에 해당하는 태그가 존재하지 않습니다.");
+        }
+    }
+
+    @Nested
+    @DisplayName("태그명으로 태그 조회")
+    class fetchByName {
+        @Test
+        @DisplayName("성공 : 태그명으로 태그를 조회할 수 있다.")
+        void fetchByNameSuccess() {
+            Tag tag = tagRepository.save(new Tag("tag"));
+
+            Tag actual = tagRepository.fetchByName(tag.getName());
+
+            assertThat(actual).isEqualTo(tag);
+        }
+
+        @Test
+        @DisplayName("실패 : 존재하지 않는 태그명인 경우 에러가 발생한다.")
+        void fetchByNameFailByNotExistsId() {
+            String notExistTagName = "태그";
+
+            assertThatThrownBy(() -> tagRepository.fetchByName(notExistTagName))
+                    .isInstanceOf(CodeZapException.class)
+                    .hasMessage("이름이 " + notExistTagName + "인 태그는 존재하지 않습니다.");
+        }
+    }
+
+    @Nested
+    @DisplayName("태그명으로 태그 조회")
+    class findByName {
+        @Test
+        @DisplayName("성공 : 태그명으로 태그를 알아낼 수 있다.")
+        void findByNameSuccess() {
+            Tag tag = tagRepository.save(new Tag("태그"));
+
+            Optional<Tag> actual = tagRepository.findByName(tag.getName());
+
+            assertThat(actual).isEqualTo(Optional.of(tag));
```

Comment:
```suggestion
            assertThat(actual).hasValue(tag);
```

이런 메서드가 있더라구요! 활용해볼까요?

Reply:
미쳤다 너무 원햇어 이 방법 ..!